### PR TITLE
#614 - check email not verified

### LINKED ISSUE #614: User login error - check email not verified first
If I try to login without verifying my email I get this error:

image

Due to:

```javascript
      const { email_verified: emailVerified } = auth0User;
      const { payload } = app.jwt.decode(token);
      const userId = payload[authConfig.jwtMongoIdKey];
      if (!userId) {
        throw new Error("no mongo_id found in JWT");
      }

```

Should we check for the 4XX error first similar to `POST users/` before the 5XX error:

```javascript
      const { email, email_verified: emailVerified } = user;
      if (!emailVerified) {
        throw app.httpErrors.forbidden("Email address not verified");
```

### PR DESCRIPTION
**Pull request by emanuelquerty**
### What does this pull request aim to achieve? 💯

This PR resolves #614 

_Please be concise and link any related issue(s):_

- [x] I have checked that no one else is working on similar changes.
- [x] I have read the latest [**README**](README.md) and understand the development workflow.
- [x] The name of this branch is: **`feature/<branch_name>`**, or **`hotfix/<branch_name>`** if this is a hotfix, where `<branch_name>` briefly describes the issue(s) resolved and is prefixed with the issue number(s) (e.g. `1127-1130-update-git-branching-model`). The branch is created in a cloned version of this repo, **not a forked repo**.
- [x] This branch is rebased/merged with the **latest staging** (or the **latest production** if it is a hotfix).
- [x] There are no merge conflicts.
- [x] There are no console warnings and errors.
- [x] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".
- [x] The title describes the issue(s) being addressed using format: **`<issue_numbers> - <brief_issue_description>`** (e.g. `#1127/#1130 - Update Git Branching Model`)

### CODE REVIEW

File: backend/lib/endpoints/auth.js [hunk 1]
```diff
@@ -124,6 +124,16 @@ async function routes(app) {
       reply.setAuthCookies(token);
       const auth0User = await Auth0.getUser(token);
       const { email_verified: emailVerified } = auth0User;
+      // Ideally, this error should be thrown to comply
```
Comment (joshmorel) [inline backend/lib/endpoints/auth.js]: Thanks @emanuelquerty . The code is great - tested it out.

The comment we want but maybe a little more concise? (2 lines max)

```
// Don't throw as already checking 403 from Auth0 (wrong email/pw)
// to respond to client as 401 unauthorized error
```

**Review by joshmorel — CHANGES_REQUESTED** [review]

**Comment by camba95** [comment]
This PR looks ok, just resolve Josh's comment

File: backend/lib/endpoints/auth.js (hunk 1 above)
Comment (emanuelquerty) [reply backend/lib/endpoints/auth.js]: Thanks for the feedback @joshmorel. I will edit the comments and make it more concise (2 lines max)

**Review by joshmorel — APPROVED** [review]
Thanks @emanuelquerty LGTM

**Review by camba95 — APPROVED** [review]

**Review by idjevm — APPROVED** [review]
LGMT! Thanks @emanuelquerty

**Comment by rabbi08** [comment]
Verified in STG, this is working fine. Please see below screenshot. Thanks.
image

**Pull request merged**